semgrep/semgrep/rule_lang.py

```diff
@@ -114,6 +114,9 @@ class Span:
     file: Optional[str]
     context_start: Optional[Position] = None
     context_end: Optional[Position] = None
+    config_path: Optional[Tuple[Any, ...]] = None
```

Comment:
What is the difference between file and config_path and start/end and config_start/end?

Reply:
config_path is the yaml path to the pattern, config_start/end are the start/end for just the pattern

Reply:
Are there errors that need to be reported on both a target file and rule file? Cant the file be used for config_path?

Reply:
No it's the path like ["rules", 0, "patterns", "pattern-either", 1]. We need it for the playground to report errors in basic mode